src/monattack.cpp

```diff
@@ -2785,14 +2786,32 @@ bool mattack::photograph( monster *z )
         }
     }
 
-    if( z->friendly ) {
-        // Friendly (hacked?) bot ignore the player.
+    if( z->friendly || g->u.weapon.typeId() == "e_handcuffs" ) {
+        // Friendly (hacked?) bot ignore the player. Arrested suspect ignored too.
         // TODO: might need to be revisited when it can target npcs.
         return false;
     }
     z->moves -= 150;
     add_msg( m_warning, _( "The %s takes your picture!" ), z->name().c_str() );
     // TODO: Make the player known to the faction
+    if( one_in( 6 ) ) {
```

Comment:
I would reduce this code block to something like this:

```CPP
    const std::string name = "...  database connection lost";
    if( one_in( 6 ) ) {
        name = Name::generate( g->u.male );
    } else if( one_in( 3 ) ) {
        name = g->u.name;
    }
    sounds::sound( z->pos(), 15, sounds::sound_t::alert, string_format( _( "a robotic voice boom, \"Citizen %s!\"" ), name ) );
```

Reply:
Done.